enhancements/network/networking-console-plugin.md

```diff
@@ -0,0 +1,148 @@
+---
+title: networking-console-plugin
+authors:
+  - "@upalatucci"
+reviewers:
+  - "@orenc1"
+  - "@zshi-redhat"
+  - "@pcbailey"
+  - "@metalice"
+  - "@spadgett"
+  - "@jhadvig"
+approvers:
+  - "@dhellmann"
+api-approvers:
+  - None
+creation-date: 2024-04-12
+last-updated: 2024-04-12
+tracking-link:
+  - https://issues.redhat.com/browse/CONSOLE-3952
+---
+
+
+# Create console plugin for the networking section
+
+## Summary
+
+At present, several pages within the networking section of our console are defined within the console repository. 
+However, working with the console repository has slowed down development and there are multiple motivations to convert static console plugins into dynamic ones described [here](https://github.com/spadgett/enhancements/blob/master/enhancements/console/dynamic-plugins.md#motivation). 
+To address this, we propose relocating the relevant code to a separate repository named "networking-console-plugin" and integrating it into the cluster-networking-operator.
```

Comment:
might be a stupid question, but why specifically do we think this should live in CNO and not in https://github.com/kubevirt/cluster-network-addons-operator ?
what would be the plugin install's impact on resource consumption on the overall ocp cluster?
when we put this in CNO its going to be ON by default; are we sure we want console on all variants of sno,ocp,hcp,microshift etc?

Reply:
Hei @tssurya. Right now the Networking console section is integrated with the console and in order to don't change the user experience, with this upgrade we would like to have this section delivered by default with openshift anyways.

So if this addons-operator is delivered by default and it's not 'optional' I would leave the decision to the CNO team.

Reply:
CNO team and adds-on team are separate, my concern is around enabling yet another daemonset lifecycle management from CNO -> resources wise what's the increase/impact here?
how does the upgrade design look like where things are currently part of console but now moving to CNO?
@upalatucci : thanks for taking the time to answer my questions; please add all this information into the enhancement so that the right context is being set (I think all the sections in the enhancements should be filled or a reasoning must be provided as to why its not relevant.).

Reply:
The Console team is stretched with a lot of work and no one gets to maintain and develop new features for network UI for years so we're taking over this part. 

For us taking over means making the code future-proof and scalable fitting our standards. 
We have already received several requests but cannot keep adding things in an already busy console code.






[11:09](https://redhat-internal.slack.com/archives/D02T26V2XC7/p1720688967509459)
And the network ui plugin place is with the network plugin it serves as a facade to

Reply:
> when we put this in CNO its going to be ON by default; are we sure we want console on all variants of sno,ocp,hcp,microshift etc?

Yes sure. We want this enabled in all places. So that when they have or add in the future the console, they can see all the Networking pages like Routes, Services, NADS and everything they already see now.

Reply:
@upalatucci thanks for summary.
From what I read we have three choices how to move forward.
1. Place the plugin directly into CNO repo (plugin will be enabled and available by default), where Ugo's team will claim ownership over the plugin code.
2. Place the plugin in the CNaddonsO repo (plugin will need to be enabled manually, since the addons need to be installed manually)
3. Splitting the logic, by placing the plugin into a separate repo and source it by the CNO or CNaddosO, based on the default behaviour thats expected (similar to how the Virtualization Operator is delivering  the kubevirt-plugin).

Here I would go with the third option, so the ownership is clearly defined.
I would suggest to tak to the kubevirt folks and check how thier operator is vendoring the plugin.